back/app/jobs/expire_confirmation_code_or_delete_job.rb

```diff
@@ -2,15 +2,9 @@
 
 class ExpireConfirmationCodeOrDeleteJob < ApplicationJob
   def run(user_id, code_to_expire)
-    # First check if the user record still exists.
     user = User.find_by(id: user_id)
-    return unless user&.confirmation_required? && user&.email_confirmation_code == code_to_expire
+    return if !user&.expire_confirmation_code!(code_to_expire)
```

Comment:
Not sure if I find this easier to understand. I always find it really confusing with these kind of negations to wrap my mind around all the possibilities (e.g. what if the user is nil?). Also, because `expire_confirmation_code!` ends with an exclamation mark, it implies it has some side effect, and it's not clear if the value it returns will be before or after it applies the side effect. And even looking at the implementation of the function, it ends with `update!`, and I don't know what that returns if anything.

My preference would be something like this. It's a bit more verbose, but for me much easier to follow:

```
return unless user
return unless user.confirmation_required?
return unless user.email_confirmation_code == code_to_expire

user.expire_confirmation_code!
```